server/src/alda/lisp/attributes.clj

```diff
@@ -14,15 +14,28 @@
   {:pre [(<= 0 x)]}
   (constantly (/ x 100.0)))
 
+;; Validation that the input is an integer value
+(defn- pos-integer[x]
+  {:pre [(and (integer? x)
+              (pos? x))]}
+  (constantly x))
+
 (defattribute tempo
   "Current tempo. Used to calculate the duration of notes."
-  :initial-val 120)
+  :initial-val 120
+  :transform pos-integer)
 
 (defattribute duration
   "Default note duration in beats."
   :initial-val 1
   :fn-name set-duration
+  ;; :aliases [:duration]
   :transform (fn [val]
+               { :pre [(or
+                        (map? val)
+                        (and (number? val)
+                             (pos? val)))]}
+
```

Comment:
1. In a Clojure map, there should usually be no space between the opening curly brace and the first thing in it.

2. Re: line breaks / spacing, this is a matter of preference, but I would write it like this:

```clojure
(fn [val]
  {:pre [(or (map? val)
             (and (number? val) (pos? val)))]}
```